ros_cross_compile/ros_cross_compile.py

```diff
@@ -174,7 +182,14 @@ def cross_compile_pipeline(
 def main():
     """Start the cross-compilation workflow."""
     args = parse_args(sys.argv[1:])
-    cross_compile_pipeline(args)
+    ros_workspace_dir = resolve_ros_workspace(args.ros_workspace)
+    data_collector = DataCollector()
+    data_writer = DataWriter(ros_workspace_dir)
+
+    with DataTimer('e2e', data_collector):
+        cross_compile_pipeline(args)
+
+    data_writer.wrap_up(data_collector)
```

Comment:
```suggestion
    try:
        with DataTimer('e2e', data_collector):
            cross_compile_pipeline(args)
    finally:
        data_writer.write(data_collector)
```

This will need to be exception safe